tests/specialization_constants/specialization_constants_multiple.h

```diff
@@ -0,0 +1,215 @@
+
+#ifndef __SYCLCTS_TESTS_SPEC_CONST_MULTIPLE_H
+#define __SYCLCTS_TESTS_SPEC_CONST_MULTIPLE_H
+
+#include "../common/common.h"
+#include "../common/once_per_unit.h"
+#include "specialization_constants_common.h"
+
+template <typename T, int def_val>
+constexpr sycl::specialization_id<T> sc_multiple(
+    get_spec_const::value_helper<T>(def_val));
+
+namespace specialization_constants_multiple {
+using namespace sycl_cts;
+using namespace get_spec_const;
+
+template <typename T, typename via_kb>
+class sc_kernel_multiple;
+
+constexpr int size = 6;
+constexpr std::array<int, size> def_values{11, 22, 33, 44, 55, 66};
+
+template <typename T, typename ResAccType>
+void get_values_from_handler(ResAccType &res_acc, sycl::kernel_handler &h) {
+  res_acc[0] = h.get_specialization_constant<sc_multiple<T, def_values[0]>>();
+  res_acc[1] = h.get_specialization_constant<sc_multiple<T, def_values[1]>>();
+  res_acc[2] = h.get_specialization_constant<sc_multiple<T, def_values[2]>>();
+  res_acc[3] = h.get_specialization_constant<sc_multiple<T, def_values[3]>>();
+  res_acc[4] = h.get_specialization_constant<sc_multiple<T, def_values[4]>>();
+  res_acc[5] = h.get_specialization_constant<sc_multiple<T, def_values[5]>>();
+}
+
+template <typename T, typename via_kb>
+class check_specialization_constants_multiple_for_type {
+ public:
+  void operator()(util::logger &log, const std::string &type_name) {
+    // Set several values via kernel_bundle, launch a kernel, and read
+    // these values and some default values from the kernel.
+    auto queue = util::get_cts_object::queue();
+    const sycl::context ctx = queue.get_context();
+    const sycl::device dev = queue.get_device();
+
+    if constexpr (via_kb::value) {
+      if (!dev.has(sycl::aspect::online_compiler)) {
+        once_per_unit::log(log, "Device does not support online compilation");
+        return;
+      }
+    }
+    int val_A = 5;
+    int val_B = 10;
+    int val_C = 30;
+    T ref1 = T(value_helper<T>(0));
+    T ref2 = T(value_helper<T>(0));
+    T ref3 = T(value_helper<T>(0));
+    init_values(ref1, val_A);
+    init_values(ref2, val_B);
+    init_values(ref3, val_C);
+    // to not initialize for struct with no default constructor
+    T *result_vec = (T *)malloc(size * sizeof(T));
+    {
+      sycl::buffer<T, 1> result_buffer(result_vec, sycl::range<1>(size));
+      queue.submit([&](sycl::handler &cgh) {
+        auto res_acc =
+            result_buffer.template get_access<sycl::access_mode::write>(cgh);
+        // Via kernel_bundle
+        if constexpr (via_kb::value) {
+          auto kernelId = sycl::get_kernel_id<sc_kernel_multiple<T, via_kb>>();
+          sycl::kernel_bundle k_bundle =
+              sycl::get_kernel_bundle<sycl::bundle_state::input>(ctx, {dev},
+                                                                {kernelId});
+            if (!k_bundle.has_kernel(kernelId)) {
+            log.note("kernel_bundle doesn't contain target kernel;"
```

Comment:
Indent

Reply:
Fix indent issue in this commit: 63a20f72785ee59bddc3ecec253f9811a63509f4